code/modules/mob/living/living.dm

```diff
@@ -1395,3 +1397,39 @@
 	if(is_servant_of_ratvar(src))
 		eminence.selected_mob = src
 		to_chat(eminence, "<span class='brass'>You select [src].</span>")
+
+
+/mob/living/ghostize(can_reenter_corpse = TRUE, retain_sentience = FALSE )
+	..()
+	if(retain_sentience)
+		set_playable()
+
+/mob/living/attack_ghost(mob/user)
+	. = ..()
+	if(.)
+		return
+	give_mind(user)
+
+/mob/living/Topic(href, href_list)
+	if(..() && href_list["activate"])
```

Comment:
this doesn't work

Reply:
I'll mark this as a draft. There's some more changes that need to be done.